src/utils/resolveAliasedImport.ts

```diff
@@ -59,35 +56,51 @@ export const resolveAliasedImport = ({
         return null;
     }
 
-    const paths = config.config?.compilerOptions?.paths as void | TsconfigPaths;
-    const baseUrl = config.config?.compilerOptions?.baseUrl as void | string;
+    const paths: unknown = config.config?.compilerOptions?.paths;
+
+    const potentialBaseUrl: unknown = config.config?.compilerOptions?.baseUrl;
+
     const configLocation = path.dirname(config.filepath);
 
-    if (!validate.string(baseUrl)) {
-        return null;
-    }
-    if (!validate.tsconfigPaths(paths)) {
-        return null;
-    }
+    if (validate.tsconfigPaths(paths)) {
+        const baseUrl = validate.string(potentialBaseUrl)
+            ? potentialBaseUrl
+            : '.';
 
-    for (const alias in paths) {
-        const aliasRe = new RegExp(alias.replace('*', '(.+)'), '');
+        for (const alias in paths) {
+            const aliasRe = new RegExp(alias.replace('*', '(.+)'), '');
 
-        const aliasMatch = importFilepath.match(aliasRe);
+            const aliasMatch = importFilepath.match(aliasRe);
 
-        if (aliasMatch == null) continue;
+            if (aliasMatch == null) continue;
 
-        for (const potentialAliasLocation of paths[alias]) {
-            const resolvedFileLocation = path.resolve(
-                configLocation,
-                baseUrl,
-                potentialAliasLocation
-                    // "./utils/*" -> "./utils/style.module.css"
-                    .replace('*', aliasMatch[1]),
-            );
+            for (const potentialAliasLocation of paths[alias]) {
+                const resolvedFileLocation = path.resolve(
+                    configLocation,
+                    baseUrl,
+                    potentialAliasLocation
+                        // "./utils/*" -> "./utils/style.module.css"
+                        .replace('*', aliasMatch[1]),
+                );
+
+                if (!fs.existsSync(resolvedFileLocation)) continue;
+
+                return resolvedFileLocation;
+            }
+        }
+    }
+
+    // fallback to baseUrl
+    // if paths is not defined, don't use the fallback for baseUrl
 
-            if (!fs.existsSync(resolvedFileLocation)) continue;
+    if (validate.string(potentialBaseUrl)) {
```

Comment:
This code now misses a case where `potentialBaseUrl` is undefined, but the module can be resolved as if baseUrl is set to `'.'`. You can lift the `baseUrl` declaration from the paths validation condition and reuse it here with out having to validate for it, then we would cover all cases. Once this change is done. I am happy to merge this PR.

Thanks for the prompt response and updating your PR 👌

Reply:
Would we want to only set baseUrl to `.` if there is a valid set of `paths`? At least in a test project I booted up, the `baseUrl: '.'` default only kicked in if there was a valid set of paths - otherwise, the import resolution was not able to resolve imports relative to tsconfig root.

Reply:
I pushed on extra commit with my interpretation of the tsconfig rules, but I'm not 100% sure on it... just what I had seen locally, the documentation is a bit fuzzy on the baseUrl behaviour when no paths are set unfortunately

Reply:
Ah, my bad. I've just tested this case with no paths and no baseUrl and you're right. Your previous commit was correct. Sorry about making you make this change. If you can roll back to your previous change and leave the comment about typescript only defaulting baseUrl to '.' with valid paths, I am happy to merge.

Sorry about this. Though glad this tricky scenario will be documented for future reference.

Reply:
Thanks for double checking! Just reverted the commit + added some comments, should be good to go now.